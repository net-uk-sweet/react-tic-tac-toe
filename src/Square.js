import React from 'react';

export default function Square(props) {
    const classes = props.highlight ? 'square highlight' : 'square';
    return (
      <button 
        className={classes}
        onClick={props.onClick}>
          {props.value}
      </button>
    );
  }