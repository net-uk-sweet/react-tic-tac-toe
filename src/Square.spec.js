import React from 'react';
import { shallow } from 'enzyme';

import Square from './Square';

let shared;
let mockedClick;

beforeEach(() => {
    mockedClick = jest.fn();
    shared = {
        props: {
            value: 'x',
            highlight: false,
            onClick: mockedClick
        }
    }
    shared.wrapper = shallow(<Square {...shared.props} />);
});

it('matches snapshot', () => {
    expect(shared.wrapper.getElement()).toMatchSnapshot();
});

it('highlights when part of a winning sequence', () => {
    shared.props.highlight = true;
    shared.wrapper = shallow(<Square {...shared.props} />).getElement();

    expect(shared.wrapper).toMatchSnapshot();
});

it('it calls the onClick callback when clicked', () => {
    shared.wrapper.find('button').simulate('click');
    expect(mockedClick).toHaveBeenCalled();
});
