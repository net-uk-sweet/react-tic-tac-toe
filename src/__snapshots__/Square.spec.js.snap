// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlights when part of a winning sequence 1`] = `
<button
  className="square highlight"
  onClick={[Function]}
>
  x
</button>
`;

exports[`matches snapshot 1`] = `
<button
  className="square"
  onClick={[Function]}
>
  x
</button>
`;
